gradio/utils.py

```diff
@@ -805,6 +811,121 @@ def __str__(self):
         return ""
 
 
+def make_waveform(
+    audio: str | Tuple[int, np.ndarray],
+    *,
+    bg_color: str = "#f3f4f6",
+    bg_image: str = None,
+    fg_alpha: float = 0.75,
+    bars_color: str | Tuple[str, str] = ("#fbbf24", "#ea580c"),
+    bar_count: int = 50,
+    bar_width: float = 0.6,
+):
+    """
+    :param audio: Audio file path or tuple of (sample_rate, audio_data)
+    :param bg_color: Background color of waveform (ignored if bg_image is provided)
+    :param bg_image: Background image of waveform
+    :param fg_alpha: Opacity of foreground waveform
+    :param bars_color: Color of waveform bars. Can be a single color or a tuple of (start_color, end_color) of gradient
+    :param bar_count: Number of bars in waveform
+    :param bar_width: Width of bars in waveform. 1 represents full width, 0.5 represents half width, etc.
+    """
+    if isinstance(audio, str):
+        audio_file = audio
+        audio = processing_utils.audio_from_file(audio)
+    else:
+        tmp_wav = tempfile.NamedTemporaryFile(suffix=".wav")
```

Comment:
Needed for Windows
```suggestion
        tmp_wav = tempfile.NamedTemporaryFile(suffix=".wav", delete=False)
```

Reply:
disagree, why should intermediate files not have delete=True? When this function ends, the files will be cleaned up by garbage collection which is what we want,